### PR TITLE
feat: Add code_signing_config_arn support

### PR DESCRIPTION
**Pull request by tsoe77**
## Description

This PR adds `code_signing_config_arn` support.

The change doesn't effect any other things except it takes in `code_signing_config_arn` if user wants to provide one.

### CODE REVIEW

**Comment by tsoe77** [comment]
Fixes #261

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks simple and good.

Please add the working example for this feature in the `examples` folder (probably making a new example folder that will create all required code signing resources and lambda function).

Also, add new outputs for `signing_job_arn`(ARN of the signing job) and `signing_profile_version_arn` (ARN of the signing profile version). I think, they are related to this change, right?

**Comment by bryantbiggs** [comment]
@tsoe77 you could probably copy a bit from https://github.com/clowdhaus/aws-lambda-code-signing-action/blob/main/__infra__/signing.tf for the new example (if looking for a setup to get started) - just FYI

**Comment by tsoe77** [comment]
Thanks both for the tips, I will be adding the examples this week 😀

**Comment by agjmills** [comment]
Would there be any opportunity to reopen this pull request @tsoe77  /  @antonbabenko ? It looks like all of the work is done and it would be very useful functionality to add to this module!

**Comment by tsoe77** [comment]
@antonbabenko if you could reopen this PR, I don't mind working on this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,59 @@
 
 All notable changes to this project will be documented in this file.
 
+### [4.0.2](https://github.com/terraform-aws-modules/terraform-aws-lambda/compare/v4.0.1...v4.0.2) (2022-09-17)
```
Comment (bryantbiggs) [inline CHANGELOG.md:5]: it looks like you have some changes to resolve - these should not be included in the changeset

Comment (tsoe77) [reply CHANGELOG.md:5]: Thanks @bryantbiggs, I will have a look and resolve them.

**Pull request closed without merging**